substrate/frame/multisig/src/lib.rs

```diff
@@ -665,6 +691,25 @@ impl<T: Config> Pallet<T> {
 		signatories.insert(index, who);
 		Ok(signatories)
 	}
+
+	pub(crate) fn ensure_updated(id: &T::AccountId, call_hash: &[u8; 32], threshold: u16) -> bool {
+		#[allow(deprecated)]
+		let r = match Multisigs::<T>::take(id, call_hash) {
+			Some(r) => r,
+			None => return false,
+		};
+		T::Currency::unreserve(&r.depositor, r.deposit);
+		// take consideration
+		let Ok(ticket) =
+			T::Consideration::new(&r.depositor, Footprint::from_parts(1, threshold as usize))
```

Comment:
when migrating, shouldn't we take exactly `r.deposit` instead of calculating using the current config? to handle the case where the reserve amounts changed.

Reply:
As we are un-reserving the previous amount, isn't it a fair assumption to charge the current amount in a migration?

Reply:
I wouldn't say so, because it's not what the user agreed to / understood at the time that they made the multisig. I would expect deposit amount changes to only impact new multisigs, not be applied retroactively

Reply:
I see your point, but I also don't think that this is a significant issue. @ggwpez @kianenigma Would love to get your thoughts here. I imagine this was the reason we were trying to add `new_from_exact` API in consideration, but this looks to me as a better trade-off.

Reply:
Yes this is why we were trying to add `new_from_exact`. I guess adding that failed/is still in progress? :P

Reply:
https://github.com/paritytech/polkadot-sdk/pull/2274 can be used to migrate them over with an exact amount

Reply:
> I think we shouldn't change the amounts on the fly, the user only agreed to the amount that they signed and if they are early adopters of Polkadot we should not punish that for no good reason.

I agree

Reply:
> I think we shouldn't change the amounts on the fly, the user only agreed to the amount that they signed and if they are early adopters of Polkadot we should not punish that for no good reason.

Thanks - but I do see this pattern already being used in preimage pallet [here](https://github.com/paritytech/polkadot-sdk/blob/f80cfc22595bbb39b50d2bd39661ff9eef74cb60/substrate/frame/preimage/src/lib.rs#L260). Won't that be impacted with a similar reasoning?

Reply:
Yes, I would have argued to not have used that pattern in the preimage pallet.